stock_picking_actual_date/models/stock_move.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2024 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    actual_date = fields.Date(
+        compute="_compute_actual_date",
+        store=True,
+    )
+
+    def _get_timezone(self):
+        return self.env.context.get("tz") or self.env.user.tz or "UTC"
+
+    @api.depends("date", "picking_id.actual_date", "scrap_ids.actual_date")
+    def _compute_actual_date(self):
+        tz = self._get_timezone()
+        for rec in self:
+            if rec.scrapped:
+                if rec.scrap_ids and rec.scrap_ids[0].actual_date:
+                    rec.actual_date = rec.scrap_ids[0].actual_date
+                continue
```

Comment:
This implementation is incorrect as it won't set actual_date on the stock move when there is no actual_date set in scrap.

I think the following should do.
```suggestion
            if rec.scrap_ids.actual_date:
                rec.actual_date = rec.scrap_ids.actual_date
                continue
```